### DISCUSSION TITLE
Dynamic Middlewares - Can't find `addRouteMiddleware`

### DISCUSSION THREAD

**Discussion by 1nVitr0** [nuxt/nuxt#17604] [Questions]
The [Nuxt3 Middleware documentation](https://v3.nuxtjs.org/docs/directory-structure/middleware) states, that you can add middlewares dynamically using plugins. Bu I can't for the life of me find the references "helper function" `addRouteMiddleware`. The code in the documentation is as follows:

```
export default defineNuxtPlugin(() => {
  addRouteMiddleware('global-test', () => {
    console.log('this global middleware was added in a plugin and will be run on every route change')
  }, { global: true })

  addRouteMiddleware('named-test', () => {
    console.log('this named middleware was added in a plugin and would override any existing middleware of the same name')
  })
})
```

Has the name of the function changed or is the process different now?

**Comment by wobsoriano** [comment]
I think it's a new feature. Try to reinstall nuxt3 again

**Reply by 1nVitr0** [reply]
I am on the newest version. Do you know where the function is supposed to be coming from ? Neither `@nuxt/kit`, nor `nuxt3` nor `#app` export the function as far as I can see. Although it could of course be untyped, I'll check that tonight.

**Reply by danielroe** [reply]
You need to have a `pages/` directory at the moment, as it's a feature of the routing integration.

**Reply by 1nVitr0** [reply]
Alright, I presume it's a typing issue then (I have a `pages` directory). Will report back as soon as I tested it

**Reply by GithubMitch** [reply]
Any news on this ?
